graylog2-server/src/main/java/org/graylog2/indexer/searches/SearchesConfig.java

```diff
@@ -87,10 +87,19 @@ public static Builder builder() {
 
         public abstract Builder limit(int limit);
 
+        public abstract int limit();
+
         public abstract Builder offset(int offset);
 
         public abstract Builder sorting(Sorting sorting);
 
-        public abstract SearchesConfig build();
+        abstract SearchesConfig autoBuild();
+
+        public SearchesConfig build() {
+            if (limit() == 0) {
```

Comment:
The condition before (line 68) was `limit <= 0`. Does that make any difference in practice?